types/messages.go

```diff
@@ -138,103 +130,206 @@ func (msg *SSVMessage) Decode(data []byte) error {
 type SignedSSVMessage struct {
-	Signature  [256]byte // Created by the operator's network key
-	OperatorID OperatorID
-	Data       []byte
+	Signatures  [][]byte     `ssz-max:"13,256"` // Created by the operators' key
+	OperatorIDs []OperatorID `ssz-max:"13"`
+	SSVMessage  *SSVMessage
+	// Full data max value is ConsensusData max value ~= 2^8 + 8 + 2^20 + 2^22 = 5243144
+	FullData []byte `ssz-max:"5243144"`
 }
 
-// GetOperatorID returns the sender operator ID
-func (msg *SignedSSVMessage) GetOperatorID() OperatorID {
-	return msg.OperatorID
+// GetOperatorIDs returns the sender operator ID
+func (msg *SignedSSVMessage) GetOperatorIDs() []OperatorID {
+	return msg.OperatorIDs
 }
 
 // GetSignature returns the signature of the OperatorID over Data
-func (msg *SignedSSVMessage) GetSignature() [256]byte {
-	return msg.Signature
+func (msg *SignedSSVMessage) GetSignature() [][]byte {
+	return msg.Signatures
 }
 
 // GetData returns message Data as byte slice
-func (msg *SignedSSVMessage) GetData() []byte {
-	return msg.Data
-}
-
-// GetSSVMessageFromData returns message SSVMessage decoded from data
-func (msg *SignedSSVMessage) GetSSVMessageFromData() (*SSVMessage, error) {
-	ssvMessage := &SSVMessage{}
-	err := ssvMessage.Decode(msg.Data)
-	if err != nil {
-		return nil, errors.Wrap(err, "could not decode SSVMessage from data in SignedSSVMessage")
-	}
-	return ssvMessage, nil
+func (msg *SignedSSVMessage) GetSSVMessage() *SSVMessage {
+	return msg.SSVMessage
 }
 
 // Encode returns a msg encoded bytes or error
 func (msg *SignedSSVMessage) Encode() ([]byte, error) {
-	return EncodeSignedSSVMessage(msg.Data, msg.OperatorID, msg.Signature), nil
+	return msg.MarshalSSZ()
 }
 
 // Decode returns error if decoding failed
 func (msg *SignedSSVMessage) Decode(data []byte) error {
-	msgData, operatorID, signature, err := DecodeSignedSSVMessage(data)
-	if err != nil {
-		return errors.Wrap(err, "could not decode data into a SignedSSVMessage")
-	}
-	msg.Data = msgData
-	msg.OperatorID = operatorID
-	msg.Signature = signature
-	return nil
+	return msg.UnmarshalSSZ(data)
+}
+
+// GetRoot returns the root
+func (msg *SignedSSVMessage) GetRoot() ([32]byte, error) {
+	return msg.HashTreeRoot()
 }
 
 // Validate checks the following rules:
-// - OperatorID should not be 0
-// - Signature length should not be 0
-// - Data length should not be 0
+// - OperatorID must have at least one element
+// - Any OperatorID must not be 0
+// - The number of signatures and OperatorIDs must be the same
+// - Any signature must not have length 0
+// - SSVMessage must not be nil
 func (msg *SignedSSVMessage) Validate() error {
-	if msg.OperatorID == 0 {
-		return errors.New("signer ID 0 not allowed")
+	// Validate OperatorID field
+	if len(msg.OperatorIDs) == 0 {
+		return errors.New("no signers")
+	}
+	// Check unique signers
+	signed := make(map[OperatorID]bool)
```

Comment:
just noting that from spec perspective we just need to make it readable and we less care about optimized memory allocation.

I guess your way is also readable and since you guys tend to copy-paste from spec I changed.